README: document how we version releases (#456)
googleapis/python-spanner-django: README.md

### Internals
![](./assets/internals.png)

> ## Versioning

Reviewer: This section seems better in CONTRIBUTING.md. The README is targeted at users who don't care about this. (It's already covered above in the statement "Use the version of django-spanner that corresponds to your version of Django.
For example, django-spanner 2.2.x works with Django 2.2.y. (This is the only supported version at this time.)")

> ## Release versioning

Reviewer 2: ```suggestion
## Versioning
```

See this [client library README](https://github.com/googleapis/nodejs-spanner/blob/831846cd86e91fdb343ac2f64a6b257877e0e157/.kokoro/presubmit/node10/system-test-against-emulator.cfg) as an example of how to explain versioning in the README.

Author: Have we yet decided if we are promoting this to alpha, beta etc? I thought for now we are documenting the basic version just to get a release.

Reviewer 2: The version documentation is independent of releases. It's just a section in the docs to show what release scheme we're using. So it'll be whatever defines the <django version><pre-release><version> format.

In any case, we'll always start with alpha. I don't think we want to publish something before alpha to pypi.

> The release version of django-spanner in [setup.py](./setup.py) SHOULD follow this form

Reviewer 2: ```suggestion
The release version of django-spanner in [setup.py](./setup.py) follows this format:
```

Author: I'd encourage we include the `SHOULD` as it is the way that django-backends must be versioned.

Reviewer 2: Ok in that case, "must" but there's no need to capitalize it.

> The release version of django-spanner in [setup.py](./setup.py) SHOULD follow this form

Reviewer 2: I would still like you to add a reference to some documentation, either a PEP or something more Django-specific that expects us to follow this convention.

I looked through [PEP 440](https://www.python.org/dev/peps/pep-0440) and I didn't find anything relevant but you may have a different reference.

Author: Could we perhaps dig up the reference after merging this PR and releasing? I got the information and guidance from Tim whom I'll consult to dig up the reference.

Reviewer 2: The release is still pending approval so if we can complete the README in the meantime, that would be great.

```python
version = '<DJANGO_VERSION><RELEASE_MODE><RELEASE_VERSION>'
```

> for example, if this package supports Django version 2.2.X, and we are in alpha release mode, while this application's release is version 0, our version will be

Reviewer 2: ```suggestion
For example, if this package supports Django version 2.2.X, and we are in alpha release mode, while this application's release is version 0, our version will be:
```

Author: Ack, thank you!

```python

> version = '2.2a0'

Reviewer: a0 means "pre-alpha" (never released). For the first alpha release, it should be a1.
